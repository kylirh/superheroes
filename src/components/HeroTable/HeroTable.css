.hero-table {
    padding: 0;
    border: 0;
    margin: 0;
    width: 100%;
    text-align: left;
    border-collapse: collapse;
}

.hero-table td {
    padding: 6px 2px;
}

.hero-table-header td {
    background: #666;
    color: #fff;
    padding: 10px 4px;
    font-weight: bold;
}

.hero-table-header td:first-child {
    border-top-left-radius: 6px;
}

.hero-table-header td:last-child {
    border-top-right-radius: 6px;
}

.hero-table-header .sortable {
    cursor: pointer;
}

.hero-table-header .sortable:hover {
    background: #444;
}
