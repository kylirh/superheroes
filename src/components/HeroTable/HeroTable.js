import React, { Component } from 'react';
import HeroRow from '../HeroRow/HeroRow'
import './HeroTable.css'

class HeroTable extends Component {
  render() {
    const rows = this.props.heroes.map(hero => <HeroRow hero={hero} key={hero.id} onSelect={this.props.onSelect} />);
    const sortBy = this.props.onSort;

    return (
        <table>
          <thead>
            <tr>
              <td></td>
              <td className="sortable" onClick={() => sortBy("id")}>ID</td>
              <td className="sortable" onClick={() => sortBy("name")}>Name</td>
              <td className="sortable" onClick={() => sortBy("persona")}>Persona</td>
              <td>Status</td>
            </tr>
          </thead>
          <tbody>{rows}</tbody>
        </table>
    );
  }
}

export default HeroTable;
