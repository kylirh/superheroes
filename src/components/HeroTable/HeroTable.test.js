import React from 'react';
import HeroTable from './HeroTable';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

describe("HeroTable", () => {
    let heroList;

    beforeEach(() => {
        heroList = [
            { id: 1, name: "Konrad Sanro", persona: "Retwoner", active: false },
            { id: 2, name: "Rowan Bredal", persona: "Hyprona", active: false },
            { id: 3, name: "Ulani Kywann", persona: "The Archkay", active: false },
            { id: 4, name: "Broadrick Ikon", persona: "E-NINJA", active: false },
            { id: 5, name: "Stuard Jedrek", persona: "Blades", active: false },
            { id: 6, name: "Bruce Wayne", persona: "Batman", active: true },
            { id: 7, name: "Clark Kent", persona: "Superman", active: true },
            { id: 8, name: "Logan", persona: "Wolverine", active: true },
            { id: 9, name: "Peter Parker", persona: "Spider Man", active: true },
            { id: 10, name: "Tony Stark", persona: "Iron Man", active: true },
        ];
    });

    it("renders correctly", () => {
        const component = shallow(
            <HeroTable heroes={heroList} />
        );
        expect(shallowToJson(component)).toMatchSnapshot();
    });

    it("calls the passed sort method", () => {
        const sortBy = jest.fn();

        const component = shallow(
            <HeroTable heroes={heroList} onSort={sortBy} />
        );

        // Sort by select checkbox
        component.find("td").at(0).simulate("click");
        expect(sortBy).not.toBeCalled();

        // Sort by status
        component.find("td").at(4).simulate("click");
        expect(sortBy).not.toBeCalled();

        // Sort by id
        component.find("td").at(1).simulate("click");
        expect(sortBy).lastCalledWith("id");

        // Sort by name
        component.find("td").at(2).simulate("click");
        expect(sortBy).lastCalledWith("name");

        // Sort by persona
        component.find("td").at(3).simulate("click");
        expect(sortBy).lastCalledWith("persona");
    });
})
