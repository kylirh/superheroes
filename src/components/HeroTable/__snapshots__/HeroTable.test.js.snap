// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeroTable renders correctly 1`] = `
<table
  className="hero-table"
>
  <thead
    className="hero-table-header"
  >
    <tr>
      <td />
      <td
        className="cell-center sortable"
        onClick={[Function]}
      >
        ID
      </td>
      <td
        className="sortable"
        onClick={[Function]}
      >
        Name
      </td>
      <td
        className="sortable"
        onClick={[Function]}
      >
        Persona
      </td>
      <td>
        Status
      </td>
    </tr>
  </thead>
  <tbody>
    <HeroRow
      hero={
        Object {
          "active": false,
          "id": 1,
          "name": "Konrad Sanro",
          "persona": "Retwoner",
        }
      }
      key="1"
    />
    <HeroRow
      hero={
        Object {
          "active": false,
          "id": 2,
          "name": "Rowan Bredal",
          "persona": "Hyprona",
        }
      }
      key="2"
    />
    <HeroRow
      hero={
        Object {
          "active": false,
          "id": 3,
          "name": "Ulani Kywann",
          "persona": "The Archkay",
        }
      }
      key="3"
    />
    <HeroRow
      hero={
        Object {
          "active": false,
          "id": 4,
          "name": "Broadrick Ikon",
          "persona": "E-NINJA",
        }
      }
      key="4"
    />
    <HeroRow
      hero={
        Object {
          "active": false,
          "id": 5,
          "name": "Stuard Jedrek",
          "persona": "Blades",
        }
      }
      key="5"
    />
    <HeroRow
      hero={
        Object {
          "active": true,
          "id": 6,
          "name": "Bruce Wayne",
          "persona": "Batman",
        }
      }
      key="6"
    />
    <HeroRow
      hero={
        Object {
          "active": true,
          "id": 7,
          "name": "Clark Kent",
          "persona": "Superman",
        }
      }
      key="7"
    />
    <HeroRow
      hero={
        Object {
          "active": true,
          "id": 8,
          "name": "Logan",
          "persona": "Wolverine",
        }
      }
      key="8"
    />
    <HeroRow
      hero={
        Object {
          "active": true,
          "id": 9,
          "name": "Peter Parker",
          "persona": "Spider Man",
        }
      }
      key="9"
    />
    <HeroRow
      hero={
        Object {
          "active": true,
          "id": 10,
          "name": "Tony Stark",
          "persona": "Iron Man",
        }
      }
      key="10"
    />
  </tbody>
</table>
`;
